### PR TITLE
AP_AHRS: make getter const and correct some assignements

### PR DESCRIPTION
**Pull request by khancyr**
Some getter weren't const 
add some constcorrectness
Correct some variable assignement

### CODE REVIEW

File: libraries/AP_AHRS/AP_AHRS_DCM.cpp [hunk 1]
```diff
@@ -3,7 +3,7 @@
  *
  *       AHRS system using DCM matrices
  *
- *       Based on DCM code by Doug Weibel, Jordi Mu�oz and Jose Julio. DIYDrones.com
+ *       Based on DCM code by Doug Weibel, Jordi Mu�oz and Jose Julio. DIYDrones.com
```
Comment (amilcarlucas) [inline libraries/AP_AHRS/AP_AHRS_DCM.cpp]: Was this change intended ?

File: libraries/AP_AHRS/AP_AHRS_DCM.cpp [hunk 2]
```diff
@@ -277,19 +274,16 @@ AP_AHRS_DCM::renorm(Vector3f const &a, Vector3f &result)
  *  to approximations rather than identities. In effect, the axes in the two frames of reference no
  *  longer describe a rigid body. Fortunately, numerical error accumulates very slowly, so it is a
  *  simple matter to stay ahead of it.
- *  We call the process of enforcing the orthogonality conditions �renormalization�.
+ *  We call the process of enforcing the orthogonality conditions �renormalization�.
```
Comment (amilcarlucas) [inline libraries/AP_AHRS/AP_AHRS_DCM.cpp]: Both the original and the new characters appear strange on the github webpage. Are these supposed to be simple quotes ?

File: libraries/AP_AHRS/AP_AHRS_DCM.cpp [hunk 3]
```diff
@@ -775,16 +767,16 @@ AP_AHRS_DCM::drift_correction(float deltat)
 #define YAW_INDEPENDENT_DRIFT_CORRECTION 0
 #if YAW_INDEPENDENT_DRIFT_CORRECTION
     // step 2 calculate earth_error_Z
-    float earth_error_Z = error.z;
+    const float earth_error_Z = error.z;
 
-    // equation 10
-    float tilt = norm(GA_e.x, GA_e.y);
+    const // equation 10
```
Comment (amilcarlucas) [inline libraries/AP_AHRS/AP_AHRS_DCM.cpp]: One const too mutch

File: libraries/AP_AHRS/AP_AHRS_NavEKF.cpp [hunk 4]
```diff
@@ -301,7 +301,7 @@ void AP_AHRS_NavEKF::update_SITL(void)
                                   radians(fdm.yawRate));
 
         for (uint8_t i=0; i<INS_MAX_INSTANCES; i++) {
-            Vector3f accel(fdm.xAccel,
+            const Vector3f accel(fdm.xAccel,
```
Comment (amilcarlucas) [inline libraries/AP_AHRS/AP_AHRS_NavEKF.cpp]: Why no {} brackets here ?

File: libraries/AP_AHRS/AP_AHRS_NavEKF.cpp [hunk 5]
```diff
@@ -326,8 +326,8 @@ void AP_AHRS_NavEKF::update_SITL(void)
             // rotate earth velocity into body frame and calculate delta position
             Matrix3f Tbn;
             Tbn.from_euler(radians(fdm.rollDeg),radians(fdm.pitchDeg),radians(fdm.yawDeg));
-            Vector3f earth_vel = Vector3f(fdm.speedN,fdm.speedE,fdm.speedD);
-            Vector3f delPos = Tbn.transposed() * (earth_vel * delTime);
+            const Vector3f earth_vel = Vector3f(fdm.speedN,fdm.speedE,fdm.speedD);
```
Comment (amilcarlucas) [inline libraries/AP_AHRS/AP_AHRS_NavEKF.cpp]: Why no {} here ?

Comment (khancyr) [reply libraries/AP_AHRS/AP_AHRS_NavEKF.cpp]: because it leads to a narrowing conversion from double to float .

**Comment by khancyr** [comment]
I correct the other two mistake

File: libraries/AP_AHRS/AP_AHRS_NavEKF.cpp (hunk 5 above)
Comment (OXINARF) [reply libraries/AP_AHRS/AP_AHRS_NavEKF.cpp]: You can call the constructor though.

Comment (khancyr) [reply libraries/AP_AHRS/AP_AHRS_NavEKF.cpp]: nice, I haven't though about that :-S . Done

File: libraries/AP_AHRS/AP_AHRS.cpp [hunk 6]
```diff
@@ -245,16 +245,14 @@ void AP_AHRS::calc_trig(const Matrix3f &rot,
                         float &cr, float &cp, float &cy,
                         float &sr, float &sp, float &sy) const
 {
-    Vector2f yaw_vector;
+    Vector2f yaw_vector{rot.a.x, rot.b.x};
```
[the remark this answers is not in the document]
Comment (khancyr) [reply libraries/AP_AHRS/AP_AHRS.cpp]: done

File: libraries/AP_AHRS/AP_AHRS_DCM.cpp [hunk 7]
```diff
@@ -155,9 +155,8 @@ AP_AHRS_DCM::reset(bool recover_eulers)
         // roll and pitch estimate
 
         // Get body frame accel vector
-        Vector3f initAccVec;
         uint8_t counter = 0;
-        initAccVec = _ins.get_accel();
+        Vector3f initAccVec = _ins.get_accel();
```
[the remark this answers is not in the document]
Comment (khancyr) [reply libraries/AP_AHRS/AP_AHRS_DCM.cpp]: done

**Comment by khancyr** [comment]
rebased and corrections based on review

**Comment by khancyr** [comment]
ok rebase on master and change the commit message

**Comment by OXINARF** [comment]
@khancyr Sorry, can you rebase again to make Travis pass?

**Comment by khancyr** [comment]
rebased on master

**Comment by OXINARF** [comment]
Merged, thanks!
